recipes/libftdi/1.x/patches/0001-cmake-targets.patch

```diff
@@ -0,0 +1,787 @@
+                 OUTPUT_VARIABLE GIT_DESCRIBE_OUTPUT
+                 RESULT_VARIABLE GIT_DESCRIBE_RESULT
++                ERROR_QUIET
+                 OUTPUT_STRIP_TRAILING_WHITESPACE
+                 )
+ if(${GIT_DESCRIBE_RESULT} STREQUAL 0)
+   set(SNAPSHOT_VERSION ${GIT_DESCRIBE_OUTPUT})
+-endif ()
++endif()
+ message(STATUS "Detected git snapshot version: ${SNAPSHOT_VERSION}")
+ 
+ configure_file(ftdi_version_i.h.in "${CMAKE_CURRENT_BINARY_DIR}/ftdi_version_i.h" @ONLY)
+ 
+ # Targets
+-set(c_sources     ${CMAKE_CURRENT_SOURCE_DIR}/ftdi.c ${CMAKE_CURRENT_SOURCE_DIR}/ftdi_stream.c CACHE INTERNAL "List of c sources" )
+-set(c_headers     ${CMAKE_CURRENT_SOURCE_DIR}/ftdi.h CACHE INTERNAL "List of c headers" )
+-
+-add_library(ftdi1 SHARED ${c_sources})
++set(c_sources
++  ftdi.c
++  ftdi_stream.c
++)
++
++set(c_headers
++  ftdi.h
++  "${CMAKE_CURRENT_BINARY_DIR}/ftdi_version_i.h"
++)
++
++add_library(ftdi1 ${c_sources} ${c_headers})
++add_library(LibFTDI::ftdi1 ALIAS ftdi1)
++target_include_directories(ftdi1 PUBLIC
++        $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
++        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
++        $<INSTALL_INTERFACE:include/libftdi1>
++        )
+ 
+ math(EXPR VERSION_FIXUP "${MAJOR_VERSION} + 1")    # Compatibility with previous releases
+ set_target_properties(ftdi1 PROPERTIES VERSION ${VERSION_FIXUP}.${MINOR_VERSION}.0 SOVERSION 2)
+ # Prevent clobbering each other during the build
+ set_target_properties ( ftdi1 PROPERTIES CLEAN_DIRECT_OUTPUT 1 )
+ 
+-
+-# Dependencies
+-target_link_libraries(ftdi1 ${LIBUSB_LIBRARIES})
++target_link_libraries(ftdi1 PRIVATE LibUSB::LibUSB)
++if (APPLE)
++    target_link_libraries(ftdi1 PRIVATE ${APPLE_CF} ${APPLE_IOK})
++endif()
+ 
+ install ( TARGETS ftdi1
+           RUNTIME DESTINATION bin
+@@ -38,17 +49,6 @@ install ( TARGETS ftdi1
+           ARCHIVE DESTINATION lib${LIB_SUFFIX}
+         )
+ 
+-if ( STATICLIBS )
+-  add_library(ftdi1-static STATIC ${c_sources})
+-  target_link_libraries(ftdi1-static ${LIBUSB_LIBRARIES})
+-  set_target_properties(ftdi1-static PROPERTIES OUTPUT_NAME "ftdi1")
+-  set_target_properties(ftdi1-static PROPERTIES CLEAN_DIRECT_OUTPUT 1)
+-  install ( TARGETS ftdi1-static
+-            ARCHIVE DESTINATION lib${LIB_SUFFIX}
+-            COMPONENT staticlibs
+-          )
+-endif ()
+-
+ install ( FILES ${c_headers}
+           DESTINATION include/${PROJECT_NAME}
+           COMPONENT headers
+diff --git a/src/ftdi.c b/src/ftdi.c
+index f5b7018..03d7b07 100644
+--- a/src/ftdi.c
++++ b/src/ftdi.c
+@@ -1328,7 +1328,7 @@ static int ftdi_to_clkbits_AM(int baudrate, unsigned long *encoded_divisor)
+    AM Type chips have only four fractional subdivisors at value[15:14]
+    for subdivisors 0, 0.5, 0.25, 0.125
+ */
+-static int ftdi_to_clkbits(int baudrate, unsigned int clk, int clk_div, unsigned long *encoded_divisor)
++static int ftdi_to_clkbits(int baudrate, int clk, int clk_div, unsigned long *encoded_divisor)
```

Comment:
Actually digging in the patch I found code changes which I was not expect ... this is beyond "fixing cmake"... 700+ lines of diff and lots of meaningless changes